### ISSUE TITLE
Unhandled Error during connection.

### ISSUE THREAD

**Issue by arthosofteq** [agebrock/tunnel-ssh#98]
I could see rejection for server
![image](https://user-images.githubusercontent.com/23403144/210355986-870abe80-d0f9-455f-b03e-1ce25d8a9f45.png)
Also there is rejection for client
![image](https://user-images.githubusercontent.com/23403144/210356106-acd0471f-0259-42cf-a8d4-7ce64e314bcc.png)

But there is no rejection on main promise in the createTunnel function
![image](https://user-images.githubusercontent.com/23403144/210356302-bea436b0-54e7-4085-abfc-7c2809281d03.png)

Basically we will see 
![image](https://user-images.githubusercontent.com/23403144/210356439-627f0f7d-af5f-4568-bd81-b049573665b2.png)
if invalid options provided which makes this lib useless for prod 

Also, not sure what is "reject" on line 36?
![image](https://user-images.githubusercontent.com/23403144/210356789-13271490-d2c1-443c-8cd5-f94c419b680b.png)

**Comment by agebrock** [comment]
Thx for the feedback. 

we will look into it  on the weekend .

**Comment by agebrock** [comment]
Hi there, we improved the error handling in 5.0.5 just now. 
Let me know if your issues are solved. 

Kind Regards
Chris
